src/D2L.Security.OAuth2/Keys/Default/LocalPublicKeyProvider.cs

```diff
@@ -20,6 +21,20 @@ IInMemoryPublicKeyCache cache
 			m_cache = cache ?? throw new ArgumentNullException( nameof( cache ) );
 		}
 
+		async Task IPublicKeyProvider.PrefetchAsync() {
+			IEnumerable<JsonWebKey> jwks = await m_publicKeyDataProvider
+				.GetAllAsync()
+				.ConfigureAwait( false );
+
+			foreach( JsonWebKey jwk in jwks ) {
+				if( m_cache.Get( PUBLIC_KEY_SOURCE, jwk.Id ) is not null ) {
+					continue;
+				}
+
+				m_cache.Set( PUBLIC_KEY_SOURCE, jwk.ToSecurityToken() );
+			}
+		}

```

Comment:
NOTE: This class is only used by the auth service for the validate route. Not planning on actually leveraging it at the moment. I didn't add tests for this (but am willing to).